aws/src/test/java/org/apache/iceberg/aws/RESTSigV4SignerTest.java

```diff
@@ -0,0 +1,114 @@
+import org.apache.iceberg.rest.HTTPClient;
+import org.apache.iceberg.rest.auth.OAuth2Util;
+import org.apache.iceberg.rest.responses.ConfigResponse;
+import org.apache.iceberg.rest.responses.OAuthTokenResponse;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.HttpResponse;
+import software.amazon.awssdk.auth.signer.internal.SignerConstant;
+
+public class RESTSigV4SignerTest {
+  private static ClientAndServer mockServer;
+  private static HTTPClient client;
+
+  @BeforeClass
+  public static void beforeClass() {
+    mockServer = ClientAndServer.startClientAndServer();
+
+    System.setProperty("aws.accessKeyId", "fake_id");
+    System.setProperty("aws.secretAccessKey", "fake_secret");
+
+    Map<String, String> properties =
+        ImmutableMap.of(
+            CatalogProperties.URI,
+            "http://localhost:" + mockServer.getLocalPort(),
+            "http.request.interceptor-impl",
+            RESTSigV4Signer.class.getName());
+    client = HTTPClient.buildFrom(properties);
+  }
+
+  @AfterClass
+  public static void afterClass() throws IOException {
+    mockServer.stop();
+    client.close();
+  }
+
+  @Test
+  public void signRequestWithoutBody() {
+    mockServer
+        .when(
+            request()
+                .withMethod("GET")
+                .withPath("/v1/config")
+                .withHeader(header(HttpHeaders.AUTHORIZATION, "AWS4-HMAC-SHA256.*")))
+        .respond(HttpResponse.response().withStatusCode(HttpStatus.SC_OK).withBody("{}"));
+
+    ConfigResponse response =
+        client.get("v1/config", ConfigResponse.class, ImmutableMap.of(), e -> {});
+    assertThat(response.defaults()).isEmpty();
+  }
+
+  @Test
+  public void signRequestWithBody() {
+    mockServer
+        .when(
+            request()
+                .withMethod("POST")
+                .withPath("/v1/oauth/token")
+                .withHeader(header(HttpHeaders.AUTHORIZATION, "AWS4-HMAC-SHA256.*"))
+                .withHeader(header(SignerConstant.X_AMZ_CONTENT_SHA256)))
+        .respond(
+            HttpResponse.response()
+                .withStatusCode(HttpStatus.SC_OK)
+                .withBody(
+                    OAuth2Util.tokenResponseToJson(
+                        OAuthTokenResponse.builder()
+                            .withToken("fake_token")
+                            .withTokenType("bearer")
+                            .withIssuedTokenType("n/a")
+                            .build())));
+
+    Map<String, String> formData = Maps.newHashMap();
+    formData.put("client_id", "asdfasd");
+    formData.put("client_secret", "asdfasdf");
+    formData.put("scope", "catalog");
+
+    OAuthTokenResponse response =
+        client.postForm(
+            "v1/oauth/token", formData, OAuthTokenResponse.class, ImmutableMap.of(), e -> {});
+    assertThat(response.token()).isNotNull();
```

Comment:
maybe there should be a few more assertions here?

```
    assertThat(response.token()).isEqualTo("fake_token");
    assertThat(response.tokenType()).isEqualTo("bearer");
    assertThat(response.issuedTokenType()).isEqualTo("n/a");
```

Reply:
That's not actually what we're testing here, so I don't think that adds value.  Really we just want to test that we got the response so that if the request doesn't post, something fails.

Reply:
I updated the assertions to be more inline with expectations